### PR TITLE
Finagle fall review blog post

### PR DESCRIPTION
**Pull request by cacoco**

### CODE REVIEW

File: source/blog/2017-10-16-fall-in-for-finagle.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+Secure Mux (smux). We've added a
+[STARTTLS](https://en.wikipedia.org/wiki/Opportunistic_TLS) command into the
+Mux protocol in order to support opportunistic TLS for ThriftMux.
+
+#### Logging Improvements
+
+We've been working hard to move
+[TwitterServer](https://github.com/twitter/twitter-server) to the [SLF4J
+API](https://) and this should be merging in the near future. Stay tuned for
```
Comment (jcrossley) [inline source/blog/2017-10-16-fall-in-for-finagle.md]: I think this is missing the url?

File: source/blog/2017-10-16-fall-in-for-finagle.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+updates and documentation on the changes.
+
+#### Scrooge Updates
+
+We've been making lots of changes to improve the usability and performance of
+Scrooge and Scrooge-generated code. This includes a set of refactorings to modernize the generated code so that it’s written in terms of Filters and Services!
+
+This will allow us to address some long standing issues like server-side response classification and correctly stating on serialization/deserialization failures.
+
+Additionally, we going to try to update to libthrift 0.10.0 from 0.5.0. This is still exploratory and we’re working to understand the ramifications.
```
Comment (jcrossley) [inline source/blog/2017-10-16-fall-in-for-finagle.md]: s/we going/we're going

File: source/blog/2017-10-16-fall-in-for-finagle.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+#### Bringing Backup Requests Back
+
+Finagle has had a
+[BackupRequestFilter](https://github.com/twitter/finagle/blob/develop/finagle-exp/src/main/scala/com/twitter/finagle/exp/BackupRequestFilter.scala)
+since [late
+2012](https://github.com/twitter/finagle/commit/526e508579309711a9c56007eff2a783826d331c)
+and evolved into its current form in [early
```
Comment (jcrossley) [inline source/blog/2017-10-16-fall-in-for-finagle.md]: and it has evolved into

File: source/blog/2017-10-16-fall-in-for-finagle.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+We are currently working towards the introduction of a Finagle ‘push’ based
+model which when coupled with a single threaded concurrency model would
+provide a more efficient and natural foundation for implementing multiplexing
+protocols.
+
+#### Application-Level Streaming Research
+
+Related to the above, Finagle has historically been dependent on
+request-response style protocols: a single request would be sent by the client
+and the server would reply with a single response. And then later multiplexing
```
Comment (jcrossley) [inline source/blog/2017-10-16-fall-in-for-finagle.md]: Refactor so sentence doesn't start with And?

File: source/blog/2017-10-16-fall-in-for-finagle.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+[4](https://github.com/twitter/util/commit/1b3ce1eddbb2a9af9fa4f97f1fc1b20ac3ee6a40))
+with the ability for users to configure their service's verbosity and to
+define a verbosity when creating a new metric.
+
+For more information see the [Util](https://twitter.github.io/util) User's Guide section on [`Leveraging
+Verbosity
+Levels`](https://twitter.github.io/util/guide/util-stats/user_guide.html#leveraging-verbosity-levels).
+
+Our internal roll-out of our new load balancer, [Deterministic
+Aperture](https://github.com/twitter/finagle/blob/finagle-6.45.0/finagle-core/src/main/scala/com/twitter/finagle/loadbalancer/aperture/DeterministicOrdering.scala) continues and we’re excited about bringing this to the wider community soon!
```
Comment (jcrossley) [inline source/blog/2017-10-16-fall-in-for-finagle.md]: Comma after Deterministic Aperture

File: source/blog/2017-10-16-fall-in-for-finagle.md [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+
+### Resiliency
+
+Transparently replacing HTTP/1.1 usage with HTTP/2 is continues internally. HTTP/2 gives you the resource reductions (a single multiplexed connection) and resiliency features (fast rolling restarts without a success rate drop) that services are already accustomed to with Mux.
```
Comment (jcrossley) [inline source/blog/2017-10-16-fall-in-for-finagle.md]: s/is continues/continues

File: source/blog/2017-10-16-fall-in-for-finagle.md [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+- Finagle's 7th "birthday" (7th anniversary of the [first public
+commits](https://github.com/twitter/finagle/tree/e04e51645374f8d958d85de384142dd00f4b7574))
+is approaching. Stay tuned for a celebratory blog post!
+
+- We're retiring the [Finatra](https://github.com/twitter/finatra)-specific
+blog, formerly available
+[here](https://twitter.github.io/finatra/blog/archives/). This will happen with 
+our next release scheduled in the next few weeks. Unfortunately, we won't be 
+keeping any of the old posts but any new information will be shared on the 
+finagle blog here.
```
Comment (kevinoliver) [inline source/blog/2017-10-16-fall-in-for-finagle.md]: s/finagle/Finagle/

File: source/blog/2017-10-16-fall-in-for-finagle.md [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+This covers the
+[August](https://github.com/twitter/finagle/tree/finagle-7.0.0) and [September](https://github.com/twitter/finagle/tree/finagle-7.1.0) releases as well as the upcoming release planned for this month. As mentioned before, we plan to continue this review
+regularly on a quarterly basis. You can start with a recap of what we were talking about [last time](https://finagle.github.io/blog/2017/07/12/summer-review/).
+
+### News & Notes
+
+------------
+
+- Finagle's 7th "birthday" (7th anniversary of the [first public
+commits](https://github.com/twitter/finagle/tree/e04e51645374f8d958d85de384142dd00f4b7574))
```
Comment (kevinoliver) [inline source/blog/2017-10-16-fall-in-for-finagle.md]: s/commits/commit/

File: source/blog/2017-10-16-fall-in-for-finagle.md (hunk 7 above)
Comment (kevinoliver) [inline source/blog/2017-10-16-fall-in-for-finagle.md]: s/finagle/Finagle/

File: source/blog/2017-10-16-fall-in-for-finagle.md (hunk 8 above)
Comment (kevinoliver) [inline source/blog/2017-10-16-fall-in-for-finagle.md]: s/commits/commit/

File: source/blog/2017-10-16-fall-in-for-finagle.md (hunk 6 above)
Comment (kevinoliver) [inline source/blog/2017-10-16-fall-in-for-finagle.md]: ffti: you could note that request cancellations no longer require tearing down the connection.

File: source/blog/2017-10-16-fall-in-for-finagle.md [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+define a verbosity when creating a new metric.
+
+For more information see the [Util](https://twitter.github.io/util) User's Guide section on [`Leveraging
+Verbosity
+Levels`](https://twitter.github.io/util/guide/util-stats/user_guide.html#leveraging-verbosity-levels).
+
+Our internal roll-out of our new load balancer, [Deterministic
+Aperture](https://github.com/twitter/finagle/blob/finagle-6.45.0/finagle-core/src/main/scala/com/twitter/finagle/loadbalancer/aperture/DeterministicOrdering.scala) continues and we’re excited about bringing this to the wider community soon!
+
+### Resiliency
```
Comment (kevinoliver) [inline source/blog/2017-10-16-fall-in-for-finagle.md]: what do you think of renaming this to New Features and then putting it up above Efficiency.

also, i think we should call out that its moved out of alpha/experimental and into the rollout phase. we could ask folks to try it if they are interested.

File: source/blog/2017-10-16-fall-in-for-finagle.md [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+Transparently replacing HTTP/1.1 usage with HTTP/2 is continues internally. HTTP/2 gives you the resource reductions (a single multiplexed connection) and resiliency features (fast rolling restarts without a success rate drop) that services are already accustomed to with Mux.
+
+### Operability
+
+We’ll be open-sourcing the work done by our summer intern [@McKardah](https://twitter.com/McKardah) to wire
+Twitter Futures into [IDEA’s async
+stacktraces](https://blog.jetbrains.com/idea/2017/02/intellij-idea-2017-1-eap-extends-debugger-with-async-stacktraces/).
+
+We’ve completed the internal rollout of [edge-triggered native
+transport](https://github.com/twitter/finagle/search?utf8=%E2%9C%93&q=nativeEpoll&type=) using Netty’s JNI edge-triggered transport, instead of using the JDK’s NIO and are working to make it available to configure externally.
```
Comment (kevinoliver) [inline source/blog/2017-10-16-fall-in-for-finagle.md]: its already available externally. we should point folks at the tunable/flag `com.twitter.finagle.netty4.UseNativeEpollV2` and recommend that they try using it.

also, i think this should move to the Efficiency section.

File: source/blog/2017-10-16-fall-in-for-finagle.md [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+configure sending a backup request to another server.
+
+#### ‘Push’ based API for Protocols
+
+Modern multiplexing protocols are characteristically very stateful and event
+driven making them challenging to implement with the current ‘pull’ based
+abstractions available in Finagle.
+
+We are currently working towards the introduction of a Finagle ‘push’ based
+model which when coupled with a single threaded concurrency model would
```
Comment (kevinoliver) [inline source/blog/2017-10-16-fall-in-for-finagle.md]: s/would/will/

File: source/blog/2017-10-16-fall-in-for-finagle.md [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+Related to the above, Finagle has historically been dependent on
+request-response style protocols: a single request would be sent by the client
+and the server would reply with a single response. And then later multiplexing
+protocols to alleviate head-of-line blocking. However, all the protocols in
+Finagle are expressed as request-response APIs with some limited support for
+streaming through using Reader and AsyncStream
+[[1](https://github.com/twitter/finagle/blob/develop/finagle-example/src/main/scala/com/twitter/finagle/example/http/HttpStreamingServer.scala)].
+These are somewhat low-level constructs and can be hard to use properly.
+
+We are investigating the design of an API for bidirectional streaming of objects not just bytes. Stay tuned.
```
Comment (bryce-anderson) [inline source/blog/2017-10-16-fall-in-for-finagle.md]: Do we need a comma after 'objects'?

File: source/blog/2017-10-16-fall-in-for-finagle.md [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+has relegated it to Finagle's experimental module.
+
+We're investigating what it would take to give users a single configuration
+variable -- ideally the percentage of additional requests to send -- to
+configure sending a backup request to another server.
+
+#### ‘Push’ based API for Protocols
+
+Modern multiplexing protocols are characteristically very stateful and event
+driven making them challenging to implement with the current ‘pull’ based
```
Comment (bryce-anderson) [inline source/blog/2017-10-16-fall-in-for-finagle.md]: "... driving _which makes_ them challenging ..." seems to flow better, at least for me.

File: source/blog/2017-10-16-fall-in-for-finagle.md (hunk 2 above)
Comment (bryce-anderson) [inline source/blog/2017-10-16-fall-in-for-finagle.md]: ffti: ".. we're going to try to upgrade libthrift from 0.5.0 to 0.10.0."

File: source/blog/2017-10-16-fall-in-for-finagle.md [hunk 14]
```diff
@@ -0,0 +1,144 @@
...
+protocols.
+
+#### Application-Level Streaming Research
+
+Related to the above, Finagle has historically been dependent on
+request-response style protocols: a single request would be sent by the client
+and the server would reply with a single response. And then later multiplexing
+protocols to alleviate head-of-line blocking. However, all the protocols in
+Finagle are expressed as request-response APIs with some limited support for
+streaming through using Reader and AsyncStream
```
Comment (isabelmartin) [inline source/blog/2017-10-16-fall-in-for-finagle.md]: ffti: s/through using/through

File: source/blog/2017-10-16-fall-in-for-finagle.md [hunk 15]
```diff
@@ -0,0 +1,144 @@
...
+[[1](https://github.com/twitter/finagle/blob/develop/finagle-example/src/main/scala/com/twitter/finagle/example/http/HttpStreamingServer.scala)].
+These are somewhat low-level constructs and can be hard to use properly.
+
+We are investigating the design of an API for bidirectional streaming of objects not just bytes. Stay tuned.
+
+#### Blocking Code Detection
+
+Our newest intern [@andrewmccree14](https://twitter.com/andrewmccree14) is
+working on a project to help detect code that is blocking in Finagle. This
```
Comment (isabelmartin) [inline source/blog/2017-10-16-fall-in-for-finagle.md]: s/code that is blocking/blocking code

File: source/blog/2017-10-16-fall-in-for-finagle.md (hunk 9 above)
Comment (cacoco) [reply source/blog/2017-10-16-fall-in-for-finagle.md]: It's hard to see here what the "this" is referring to move? The deterministic aperture section or all the of the resiliency section? Sorry.

File: source/blog/2017-10-16-fall-in-for-finagle.md (hunk 2 above)
Comment (cacoco) [reply source/blog/2017-10-16-fall-in-for-finagle.md]: Ha, that was Taylor in the google doc.

File: source/blog/2017-10-16-fall-in-for-finagle.md (hunk 12 above)
Comment (cacoco) [reply source/blog/2017-10-16-fall-in-for-finagle.md]: I was told a long time ago that when you think you need a comma, you don't.

File: source/blog/2017-10-16-fall-in-for-finagle.md (hunk 15 above)
Comment (cacoco) [reply source/blog/2017-10-16-fall-in-for-finagle.md]: That's just repeating the section title!

Comment (isabelmartin) [reply source/blog/2017-10-16-fall-in-for-finagle.md]: Ha, fair enough :)

**Comment by yufangong** [comment]
🚀

**Pull request merged**